services/eventService/eventService.go

```diff
@@ -1,43 +1,250 @@
+	}
+	return fmt.Sprintf("%d attended", e.AttendingCount)
+}
+
+func (e *Event) GetShortDate() string {
+	if e.StartTime.Year() == time.Now().Year() && e.StartTime.After(time.Now()) {
+		startTime := e.StartTime.Format("January 2")
+		if e.StartTime.YearDay() != e.EndTime.YearDay() {
+			if e.StartTime.Month() == e.EndTime.Month() {
+				return fmt.Sprintf("%s - %s", startTime, e.EndTime.Format("2"))
+			}
+			var endTime string
+
+			if e.EndTime.Year() == time.Now().Year() {
+				endTime = e.EndTime.Format("January 2")
+			} else {
+				endTime = e.EndTime.Format("January 2, 2006")
+			}
+
+			return fmt.Sprintf("%s - %s", startTime, endTime)
+		} else {
+			return startTime
+		}
+	}
+	return e.StartTime.Format("January 2, 2006")
 }
 
-// GetEvents gets the events.
-func GetEvents() []*Event {
-	events := []*Event{
-		&Event{"GreatUniHack 2016", "A first test", "Kilburn Building", "September 16 at 5:30pm", "https://google.com/", "https://www.google.co.uk/images/branding/product/ico/googleg_lodp.ico", true},
-		&Event{"Test 2", "A second test", "Here", "June 1 at 12:00pm", "https://google.com/", "https://www.google.co.uk/images/branding/product/ico/googleg_lodp.ico", false},
-		&Event{"Test 3", "A third test", "Here", "May 25 at 10:00am", "https://google.com/", "https://www.google.co.uk/images/branding/product/ico/googleg_lodp.ico", false},
+const timeFormatLayout = "2006-01-02T15:04:05-0700"
+
+var accessToken string
+
+var eventsCache *cache.Cache
+
+func init() {
+	if os.Getenv("FB_APP_ID") == "" {
+		log.Println("Environment variable FB_APP_ID is not assigned.")
+		return
 	}
 
-	return events
+	if os.Getenv("FB_SECRET") == "" {
+		log.Println("Environment variable FB_SECRET is not assigned.")
+		return
+	}
+
+	accessToken = getAccessToken()
+	eventsCache = cache.New(1*time.Minute, 0)
+
+	loadEvents()
 }
 
-// GetUpcomingEvents gets the upcoming events.
-func GetUpcomingEvents() []*Event {
-	var upcomingEvents []*Event
+// getAccessToken gets a suitable Facebook GraphAPI access token.
+func getAccessToken() string {
+	accessToken := fmt.Sprintf("%s|%s", os.Getenv("FB_APP_ID"), os.Getenv("FB_SECRET"))
+
+	return accessToken
+}
+
+func requestWithData(method, url string, data map[string]string, useToken bool) *http.Request {
+	request, _ := http.NewRequest(method, url, nil)
+
+	query := request.URL.Query()
+	for key, value := range data {
+		query.Add(key, value)
+	}
+
+	if useToken {
+		request.Header.Add("Authorization", "Bearer "+accessToken)
+	}
+
+	request.URL.RawQuery = query.Encode()
+
+	return request
+}
+
+const MAX_SHORTNAME_LENGTH = 30
+
+// getEvents gets the events.
+func getEvents() ([]*Event, error) {
+
+	req := requestWithData("GET",
+		"https://graph.facebook.com/v2.12/hacksocmcr/events", map[string]string{
+			"fields": "name,description,cover,place,start_time,end_time,attending_count,interested_count",
+		}, true)
+
+	client := &http.Client{}
+	response, err := client.Do(req)
+
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
This error handling is not complete as `client.Do` will not cause an error if there is a non-2xx status. Another if, checking for a non-2xx status is required.

Reply:
Agree

Reply:
Implemented :+1: